src/lib389/lib389/cli_idm/role.py

```
@@ -85,7 +85,7 @@ def modify(inst, basedn, log, args, warn=True):
 
 def rename(inst, basedn, log, args, warn=True):
     dn = _get_dn_arg(args.dn, msg="Enter dn to modify")
-    _generic_rename_dn(inst, basedn, log.getChild('_generic_rename_dn'), MANY, dn, args)
+    _generic_rename_dn(inst, basedn, log.getChild('_generic_rename_dn'), Roles, dn, args)
```

I'm a bit confused. `MANY == Roles` - at least, it should be, as the approach is used everywhere in our CLI.

---

> I'm a bit confused. `MANY == Roles` - at least, it should be, as the approach is used everywhere in our CLI.

Well this has been needed to fix many other bugs around this area.  It comes down to setting "self._protected = False" otherwise with "MANY" deletes are silently ignored as protected remains True for these objects

---

Sorry, and "renames", not just "deletes", also check this "protected" variable and will silently be ignored when set to True.

---

What I mean is that the change in `src/lib389/lib389/cli_idm/role.py` doesn't change anything, unless I miss something...

In the beginning of the file:
```
MANY = Roles
SINGULAR = Role
```

And this PR doesn't change anything around `protected`.
But the `xfail` removal makes sense to me.

---

Ahhh ok let me double check it, I had written this particular fix before other PR's (around Roles) got merged.

---

Yeah this issue got "accidentally" fixed in a previous PR, so the original change here was never needed.  Revised PR, and will merge.  Thanks!